zio-http/src/main/scala/zhttp/service/Client.scala

```diff
@@ -167,9 +180,15 @@ object Client {
     } yield res
   }
 
-  case class Config(socketApp: Option[SocketApp[Any]] = None, ssl: Option[ClientSSLOptions] = None) { self =>
+  case class Config(
+    socketApp: Option[SocketApp[Any]] = None,
+    ssl: Option[ClientSSLOptions] = None,
+    proxyAddress: Option[InetSocketAddress] = None,
```

Comment:
May be we want to add a new type `Proxy`. Where we can pass, `username` `password` and `headers`. They could be optional. Then we can add build a new `HttpProxyHandler` using `Proxy` class.